geonode_mapstore_client/client/js/epics/gnviewer.js

```diff
@@ -144,7 +148,34 @@ export const gnViewerRequestGeoStoryConfig = (action$) =>
                 return Observable.empty();
             });
         });
-
+export const gnViewerRequestNewGeoStoryConfig = (action$, { getState = () => {}}) =>
+    action$.ofType(REQUEST_NEW_GEOSTORY_CONFIG)
+        .switchMap(() => {
+            return Observable.defer(() => axios.all([
+                getNewGeoStoryConfig()
+            ])).switchMap((response) => {
+                const [gnGeoStory] = response;
+                const canAddResource = getState()?.security?.user?.perms.includes('add_resource');
+                if (!canAddResource) {
+                    return Observable.of(
+                        setGeoStoryResource({
+                            canEdit: false
+                        }),
+                        errorNotification({title: "geostory.errors.loading.title", message: "viewer.errors.noPermissions"})
+                    );
+                }
```

Comment:
We could move this just before `Observable.defer` so we avoid to request the base geostory if we have not permissions

Reply:
Yeah makes sense